mcpserver/tools.go

```diff
@@ -573,46 +609,55 @@ var EnvironmentFileReadTool = &Tool{
 	},
 }
 
-var EnvironmentFileListTool = &Tool{
-	Definition: mcp.NewTool("environment_file_list",
-		mcp.WithDescription("List the contents of a directory"),
-		mcp.WithString("explanation",
-			mcp.Description("One sentence explanation for why this directory is being listed."),
-		),
-		mcp.WithString("environment_source",
-			mcp.Description("Absolute path to the source git repository for the environment."),
-			mcp.Required(),
-		),
-		mcp.WithString("environment_id",
-			mcp.Description("The ID of the environment for this command. Must call `environment_create` first."),
-			mcp.Required(),
-		),
+type EnvironmentFileListToolRequest struct {
+	BaseEnvironmentRequest
+
+	Path string `json:"path"`
+}
+
+var EnvironmentFileListTool = &Tool[EnvironmentFileListToolRequest, string]{
+	Definition: newEnvironmentTool(
+		"environment_file_list",
+		"List the contents of a directory",
 		mcp.WithString("path",
 			mcp.Description("Path of the directory to list contents of, absolute or relative to the workdir"),
 			mcp.Required(),
 		),
 	),
-	Handler: func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
-		_, env, err := openEnvironment(ctx, request)
+	ParseRequest: func(request mcp.CallToolRequest) (*EnvironmentFileListToolRequest, error) {
+		base, err := parseBaseEnvironmentRequest(request)
 		if err != nil {
-			return mcp.NewToolResultErrorFromErr("unable to open the environment", err), nil
+			return nil, err
 		}
 
 		path, err := request.RequireString("path")
```

Comment:
🐛 **Bug (Medium):** In EnvironmentFileListTool ParseRequest, the path parameter extraction and error handling is duplicated. This should be moved to the struct initialization for consistency.

```suggestion
		path, err := request.RequireString("path")
		if err != nil {
			return nil, err
		}
```